From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jan Maslak <jan.maslak@intel.com>
Date: Fri, 19 Sep 2025 14:43:28 +0000
Subject: fixup! drm/xe/vma: Modify new_vma to accept struct
 xe_vma_mem_attr as parameter

Inside xe_vma_create() assert that attr isn't NULL, since it's
dereferenced later on in the function.

Signed-off-by: Jan Maslak <jan.maslak@intel.com>

(cherry-picked from commit ab95eb14fedc2c16258d4e21c43370193c9f6721 drm-tip-eudebug)
Signed-off-by: Bommu Krishnaiah <krishnaiah.bommu@intel.com>

---
 drivers/gpu/drm/xe/xe_vm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/xe/xe_vm.c b/drivers/gpu/drm/xe/xe_vm.c
index d44cdfcab3c3..25449afa4c4e 100644
--- a/drivers/gpu/drm/xe/xe_vm.c
+++ b/drivers/gpu/drm/xe/xe_vm.c
@@ -1096,6 +1096,7 @@ static struct xe_vma *xe_vma_create(struct xe_vm *vm,
 
 	xe_assert(vm->xe, start < end);
 	xe_assert(vm->xe, end < vm->size);
+	xe_assert(vm->xe, attr);
 
 	/*
 	 * Allocate and ensure that the xe_vma_is_userptr() return
-- 
2.43.0

